sdk/python/kfp/v2/compiler/compiler.py

```diff
@@ -0,0 +1,843 @@
+        pipeline,
+        root_group,
+        op_name_to_parent_groups,
+        opgroup_name_to_parent_groups,
+        condition_params,
+        op_name_to_for_loop_op,
+        pipeline_spec,
+    )
+    dependencies = self._get_dependencies(
+        pipeline,
+        root_group,
+        op_name_to_parent_groups,
+        opgroup_name_to_parent_groups,
+        opsgroups,
+        condition_params,
+    )
+
+    templates = []
+    for opsgroup in opsgroups.keys():
+      template = self._group_to_dag_template(opsgroups[opsgroup], inputs, outputs, dependencies)
+      templates.append(template)
+
+    deployment_config = pipeline_spec_pb2.PipelineDeploymentConfig()
+    importer_tasks = []
+
+
+    def _get_input_artifact_type(
+        input_name: str,
+        component_spec: ComponentSpec,
+    ) -> str:
+      """Find the input artifact type by input name."""
+      for input in component_spec.inputs:
+        if input.name == input_name:
+          return ir_types._artifact_types_mapping.get(input.type.lower()) or ''
+      return ''
+
+
+    for op in pipeline.ops.values():
+      templates.extend(op_to_templates_handler(op))
+      component_spec = op._metadata
+      task = pipeline_spec.tasks.add()
+      task.CopyFrom(op.task_spec)
+      deployment_config.executors[task.task_info.name].container.CopyFrom(op.container_spec)
+
+      # Check if need to insert importer node
+      for input_name in task.inputs.artifacts:
+        if task.inputs.artifacts[input_name].producer_task == '':
+          artifact_type = _get_input_artifact_type(input_name, component_spec)
+
+          from kfp.v2.compiler import importer_node
+          importer_task, importer_spec = importer_node.build_importer_spec(
+              task, input_name, artifact_type)
+          importer_tasks.append(importer_task)
+
+          task.inputs.artifacts[input_name].producer_task = importer_task.task_info.name
+          task.inputs.artifacts[input_name].output_artifact_key = importer_node._OUTPUT_KEY
+
+          deployment_config.executors[importer_task.executor_label].importer.CopyFrom(importer_spec)
+
+    pipeline_spec.deployment_config.Pack(deployment_config)
+
+    if len(importer_tasks) > 0:
+      pipeline_spec.tasks.extend(importer_tasks)
+
+    return templates
+
+
+  def _create_pipeline_spec(self,
+                            args: List[dsl.PipelineParam],
+                            pipeline: dsl.Pipeline,
+                            ) -> pipeline_spec_pb2.PipelineSpec:
+    """Create the pipeline spec object."""
+
+    pipeline_spec = pipeline_spec_pb2.PipelineSpec()
+    pipeline_spec.pipeline_info.name = pipeline.name or 'Pipeline'
+    pipeline_spec.sdk_version = "kfp.v2"
+    pipeline_spec.schema_version = "dummy.v1"
+
+    # Pipeline Parameters
+    for arg in args:
+      if arg.value is not None:
+        if isinstance(arg.value, int):
+          pipeline_spec.runtime_parameters[arg.name].type = pipeline_spec_pb2.PrimitiveType().INT
```

Comment:
IIRC the following should also work
```suggestion
          pipeline_spec.runtime_parameters[arg.name].type = pipeline_spec_pb2.PrimitiveType.INT
```